File: php/bob/Bob.php
<?php

declare(strict_types=1);

class Bob
{
    public function respondTo(string $str): string
    {
        $str = trim($str);

        $is_question = fn ($s) => str_ends_with($s, '?');
        $is_shouting = fn ($s) => strtoupper($s) === $s;
        $has_letters = fn ($s) => array_any(str_split($s), fn($ch) => ctype_alpha($ch));
        $is_white_space = fn ($s) => array_all(str_split($s), fn($ch) => ctype_space($ch));

        return match(true) {
            $is_white_space($str) => 'Fine. Be that way!',
            ($is_question($str) && $is_shouting($str) && $has_letters($str)) => "Calm down, I know what I'm doing!",
            $is_question($str) => 'Sure.',
            ($is_shouting($str) && $has_letters($str)) => 'Whoa, chill out!',
            default => 'Whatever.'
        };

    }
}

File: lua/bob/bob.lua
local bob = {}

function bob.hey(say)
  local is_question = say:match('?%s*$')
  local is_yelling = say == say:upper() and say:match('%a')
  local is_whitespace = say:match('^%s+$') or say == ""

  if is_question and is_yelling then
    return "Calm down, I know what I'm doing!"
  elseif is_question then
    return "Sure."
  elseif is_yelling then
    return "Whoa, chill out!"
  elseif is_whitespace then
    return "Fine. Be that way!"
  else
    return "Whatever."
  end
end

return bob
